pgosupport.cmake

```diff
@@ -37,6 +37,18 @@ function(add_pgo TargetName)
     endforeach(ConfigType)
 endfunction(add_pgo)
 
+set(CLR_CMAKE_OPTDATA_PACKAGEID "optimization.PGO.CoreCLR")
+set(CLR_CMAKE_OPTDATA_PACKAGEWITHRID "optimization.${CLR_CMAKE_TARGET_OS}-${CLR_CMAKE_TARGET_ARCH}.PGO.CoreCLR")
+
+# Parse optdata package version from project.json
+file(TO_NATIVE_PATH "${CMAKE_SOURCE_DIR}/extract-from-json.py" ExtractFromJsonScript)
+file(TO_NATIVE_PATH "${CMAKE_SOURCE_DIR}/src/.nuget/optdata/project.json" OptDataProjectJsonPath)
+execute_process(
```

Comment:
The problem is that I need to consume the output of this package at configure time. The python script is there simply to parse project.json to extract the package version, which is used to build up the ProfilePath cmake variable, which in turn is passed to the compiler/linker as a flag that tells PGO where to find the profile data.

I thought about putting the version into a text file instead of a project.json, so that I could parse the version with native cmake, but that would just move complexity into the build.cmd/run.cmd/build.proj suite of tools instead (I would have to construct a dummy project.json so that I had something valid to restore).

Is there a better approach to getting the version out of a project.json at configure time?